spring-integration-core/src/main/java/org/springframework/integration/support/management/MetricsCaptorLoader.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management;
+
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.integration.support.management.micrometer.MicrometerMetricsCaptor;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Iterates over a map of classes to check for presence (key) and classes to try to load
+ * (value) (if the key is on the classpath). The class to load must be a
+ * {@link MetricsCaptor} and have a static method {@code loadCaptor(ApplicationContext)}.
+ * User code can add to or replace the default {@link MicrometerMetricsCaptor}.
+ *
+ * @author Gary Russell
+ * @since 5.1
+ *
+ */
+public final class MetricsCaptorLoader {
```

Comment:
Why did you decide to introduce this class?

I agree that it would be better do not mix concerns, but now we have more code around `ClassUtils.isPresent()` and it's so complicated.

Thanks

Reply:
I just thought that, since I was externalizing it, it could be more configurable, but yes; agree, will simplify.